backend/mlarchive/tests/archive/actions.py

```diff
@@ -1,69 +1,43 @@
     assert 'pytest' in messages.first().get_file_path()
 
 
+@patch('mlarchive.archive.tasks.remove_selected_task.delay')
 @pytest.mark.django_db(transaction=True)
-def test_get_mbox_updates(messages):
-    apple = EmailList.objects.get(name='apple')
-    print(apple.message_set.count())
-    result = get_mbox_updates(apple.message_set.all())
-    assert result == [(1, 2017, apple.pk)]
-
-
-@patch('mlarchive.archive.tasks.update_mbox.delay')
-@pytest.mark.django_db(transaction=True)
-def test_remove_selected(mock_update, admin_user):
+def test_remove_selected(mock_update, rf, admin_user, messages):
+    '''Simple test of function. See tests/archive/utils.py for test
+    of underlying functionality'''
     mock_update.return_value = 1
-    elist = EmailListFactory.create(name='public')
-    thread = ThreadFactory.create()
-    msg = MessageFactory.create(email_list=elist, thread=thread)
-
-    # create message file
-    path = msg.get_file_path()
-    # print(path)
-    if not os.path.exists(os.path.dirname(path)):
-        os.makedirs(os.path.dirname(path))
-    with open(path, 'w') as f:
-        f.write('test.')
-
-    # ensure message file doesn't already exist in removed directory
-    target = os.path.join(msg.get_removed_dir(), msg.hashcode)
-    # print('target: {}'.format(target))
-    if os.path.exists(target):
-        os.remove(target)
-
-    query = Message.objects.all()
-    assert query.count() == 1
-    request = get_request(user=admin_user)
-    result = remove_selected(request, query)
-
-    assert result.status_code == 302
-    assert os.path.exists(target)
-    assert not os.path.exists(path)
-    assert Message.objects.count() == 0
+    request = rf.post(
+        '/arch/admin/',
+        data={'action': 'remove_selected'},
+        HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+    request.user = admin_user
+    response = remove_selected(request, messages)
+    assert isinstance(response, JsonResponse)
+    assert response.status_code == 200
 
 
+@patch('mlarchive.archive.tasks.mark_not_spam_task.delay')
 @pytest.mark.django_db(transaction=True)
-def test_not_spam(admin_user):
-    elist = EmailListFactory.create(name='public')
-    thread = ThreadFactory.create()
-    msg = MessageFactory.create(email_list=elist, thread=thread, spam_score=1)
-
-    query = Message.objects.all()
-    assert query.count() == 1
-    request = get_request(user=admin_user)
-    not_spam(request, query)
-
-    msg = Message.objects.first()
-    assert msg.spam_score == -1
+def test_not_spam(mock, rf, admin_user, messages):
+    '''Simple test of function. See tests/archive/utils.py for test
+    of underlying functionality'''
+    mock.return_value = 1
+    request = rf.post(
+        '/arch/admin/',
+        data={'action': 'not_spam'},
+        HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+    request.user = admin_user
+    response = not_spam(request, messages)
+    assert isinstance(response, JsonResponse)
+    assert response.status_code == 200
```

Comment:
Check that `mock.called` is true?

Reply:
Fixed in commit 38b59cb95e48a2c975295fe390800ddb8ef24e18